File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        Arrays.sort(nums);
         int i=0;
         int j=nums.length-1;
         HashSet<Double>value =new HashSet<>();
         while(i<j)
         {
             value.add((nums[i]+nums[j])/2.0);
             i++;
             j--;
         }
         return value.size();
    }
}

File: 2465-number-of-distinct-averages/2465-number-of-distinct-averages.cpp
class Solution {
public:
    int distinctAverages(vector<int>& nums) {
         sort(nums.begin(),nums.end());
         int i=0;
         int j=nums.size()-1;
         set<float>value;
         while(i<j)
         {
             value.insert((nums[i]+nums[j])/2.0);
             i++;
             j--;
         }
         return value.size();
    }
};